### ISSUE TITLE
Suggestion : No rescue without class

### ISSUE THREAD

**Issue by coding-bunny** [bbatsov/rubocop#2943]
Hello,

just a small suggestion.
Would it be possible to create a cop that checks rescue statements and if they are actually catching a specific error?

This would be accepted:

```
def some_function
  begin
    do_something
  rescue StandardError => e
    do_something_else
  end
end

def some_function
  do_something
rescue StandardError => e
  do_something_else
end
```

And these 2 examples would not be accepted anymore:

```
def some_function
  begin
    do_something
  rescue
    do_something_else
  end
end

def some_function
  do_something
rescue
  do_something_else
end
```

It's kinda a policy we have in our team that specific errors should be caught and not just every globbed up.

**Comment by alexdowad** [comment]
We already have something like this. `Lint/RescueException` flags `rescue Exception`, since it "swallows" *all* exceptions.

**Comment by NekoNova** [comment]
does that cover simply writing `rescue` without a class? cause just `rescue` defaults to `StandardError`

**Comment by alexdowad** [comment]
Maintainers, do we want something like this? There are a lot of Ruby codebases which use idioms like:

`a = Integer(str) rescue nil`

**Comment by bbatsov** [comment]
@alexdowad I think we have a cop that checks for this very same thing. :-) I think `RescueException` can be extended to handle @NekoNova's case as this is more or less the same thing.

**Comment by HParker** [comment]
I looked into this briefly. From what I found the `RescueException` cop is there to prevent catching exceptions that you maybe shouldn't be catching. The suggestion in this issue is that `rescue` blocks should communicate more about what type of exception they are trying to catch.

To me that sounds like a different cop. Does that sound right to others as well?

**Comment by jonas054** [comment]
@HParker That sounds right. The message from `Lint/RescueException` is "Avoid rescuing the Exception class. Perhaps you meant to rescue StandardError?". Reporting where `StandardError` is caught without being explicitly named is a different issue. So, a new cop.

**Comment by Drenmi** [comment]
@HParker: Agree with both of you on that. 😀

**Comment by olegantonyan** [comment]
Why `rescue StandardError` is ok, but just `rescue` is not if these two are the same? I would expect either `rescue StandardError` the same offense, or not to consider `rescue` without class an offense at all. It's just a syntactic sugar in Ruby.

**Comment by coding-bunny** [comment]
it's about being explicit and forcing people to think about what they want to rescue instead of gobbling up all errors.
If you catch an exception, you should handle it and do something with it, not just gobble it up.

That was the main idea behind my feature request for this.

**Comment by Drenmi** [comment]
The difference is saying "I'm deliberately rescuing `StandardError`, just so you know" vs. saying "I might have no idea what I'm rescuing, or I might, you figure it out." 🙂
